File: matrix_input.cpp
#include <iostream>
#include <vector>

int main() {
	int row, col;
	std::cout << "Enter the size of the row :";
	std::cin >> row;

	std::cout << "Enter the size of the col :";
	std::cin >> col;

	std::vector<std::vector<int>> matrix1(row, std::vector<int>(col));
	std::vector<std::vector<int>> matrix2(row, std::vector<int>(col));

	std::cout << "matrix1 Input : \n";
	for (int i = 0; i < row; i++) {
		for (int j = 0; j < col; j++) {
			std::cin >> matrix1[i][j];
		}
	}
	
	std::cout << "matrix2 Input : \n";
	for (int i = 0; i < row; i++) {
		for(int j = 0; j < col; j++) {
			std::cin >> matrix2[i][j];
		}
	}

	std::cout << "matrix1 + matrix2 result: \n";
	for (int i = 0; i < row; i++) {
		for (int j = 0; j < col; j++) {
			std::cout << matrix1[i][j] + matrix2[i][j] << " " ;
		}
		std::cout << "\n";
	}

	return 0;
}

File: matrix_input.rs
use std::io;

fn main() {
    let mut input = String::new();

    println!("Enter the size of the row: ");
    io::stdin().read_line(&mut input).expect("input error");
    let row: usize = input.trim().parse().expect("Error converting to integer");
    input.clear();

    println!("Enter the size of the col: ");
    io::stdin().read_line(&mut input).expect("input error");
    let col: usize = input.trim().parse().expect("Error converting to integer");

    let mut matrix1 = vec![vec![0; col]; row];
    let mut matrix2 = vec![vec![0; col]; row];

    println!("matrix1 input: ");
    for i in 0..row {
        input.clear();
        io::stdin().read_line(&mut input).expect("input error");
        let values: Vec<i32> = input
            .split_whitespace()
            .filter_map(|s| s.parse().ok())
            .collect();
        matrix1[i].copy_from_slice(&values);
    }

    println!("marix2 input: ");
    for i in 0..row {
        input.clear();
        io::stdin().read_line(&mut input).expect("input error");
        let values: Vec<i32> = input
            .split_whitespace()
            .filter_map(|s| s.parse().ok())
            .collect();
        matrix2[i].copy_from_slice(&values);
    }

    println!("matrix1 + matrix2 result: ");
    for i in 0..row {
        for j in 0..col {
            print!("{}", matrix1[i][j] + matrix2[i][j]);
        }
        println!();
    }
}

